data-prepper-plugins/otel-trace-source/src/main/java/com/amazon/dataprepper/plugins/source/oteltrace/OTelTraceSourceConfig.java

```diff
@@ -87,6 +91,9 @@
     @JsonProperty("authentication")
     private PluginModel authentication;
 
+    @JsonProperty(RECORD_TYPE)
+    private String recordType = DEFAULT_RECORD_TYPE;
```

Comment:
You can make this be the enum `RecordType` by using the `@JacksonCreator` approach I outlined in another comment.

Reply:
I did have a thought about the similar approach, just not sure if the traceback message for unsupported recordType thrown by Jackson will be helpful to user.